File: cheaters/PacketLife/nitro-python-1.0/apidoc/docs/samples/ns-java-sample/rewrite.java
package NSConfig;
import java.io.*;
import java.util.*;
import java.net.*;
import java.beans.*;
import java.net.*;
import NSConfig.*;
import org.apache.axis.types.*;

public class rewrite  {
	public static void main(String[] args) {
	   String  soap_server;
	   String username;
	   String password ;

	   NSConfigBindingStub client ;
	   SimpleResult  result;
	   UnsignedInt NULL_UINT=null;
	   if (args.length < 3) {
	      System.out.println("Usage : java setConfig.class <NP> <username> <password>");
	      return;
	   }
	   soap_server = args[0];
	   username    = args[1];
	   password    = args[2];
	   try {
            			// Initialized the client stub
				soap_server = "http://"+soap_server+"/soap";
				System.out.println("Connecting to  "+soap_server + " ........");

				client  = new NSConfigBindingStub(new URL(soap_server),null);
			    client.setMaintainSession(true);
				((org.apache.axis.client.Stub)client)._setProperty(org.apache.axis.AxisEngine.PROP_DOMULTIREFS, Boolean.FALSE);
				//Login
				result = client.login(username, password);
				System.out.println("Login : "+result.getMessage());

				result = client.enablensfeature((FeatureEnum)FeatureEnum.REWRITE);
				System.out.println("Enablensfeature	REWRITE : " + result.getMessage());

				result = client.addrewriteaction("rewrite_act1",(ActionrewritetypeEnum)ActionrewritetypeEnum.DELETE,"HTTP.RES.HEADER(\"Server\").VALUE(0)",null,(YesnoEnum)YesnoEnum.VALNOTSET);
				System.out.println("Addrewriteaction	rewrite_act1 : " + result.getMessage());

				result = client.addrewritepolicy("rewrite_pol1","HTTP.RES.HEADER(\"Server\").CONTAINS(\"IIS\")","rewrite_act1",null);
				System.out.println("Addrewritepolicy	rewrite_pol1 : " + result.getMessage());

				result = client.bindrewriteglobal_policy("rewrite_pol1",new UnsignedInt(1),"END",(RwglobalbindpointEnum)RwglobalbindpointEnum.VALNOTSET,false,(PolicylabelinvoketypeEnum)PolicylabelinvoketypeEnum.VALNOTSET,null);
				System.out.println("Bindrewriteglobal_policy	rewrite_pol1 : " + result.getMessage());

				result = client.logout();
				System.out.println("logout : " + result.getMessage());
	   }
	   catch (Exception ex) {
	       ex.printStackTrace();
	   }
	}

}

File: cheaters/PacketLife/nitro-python-1.0/apidoc/docs/samples/ns-csharp-sample/rewrite.cs
using System;

using System.Runtime.Remoting;

using System.Xml.Serialization;

using System.Web.Services;

    

	namespace NSConfig {

		public class ClientService : NSConfigService



                {

                        private string cookie = null;

                        /* override the getWebRequest to send cookie */

                        protected override System.Net.WebRequest  GetWebRequest(Uri uri)

                        {

                                System.Net.HttpWebRequest req = (System.Net.HttpWebRequest) base.GetWebRequest(uri);

                                if (cookie != null)

                                {

                                        req.Headers.Add("Set-Cookie", cookie);

                                }

                                return req;

                        }

                        /* override the getWebResponse to get the cookie */



                        protected override System.Net.WebResponse GetWebResponse(System.Net.WebRequest req)

                        {

                               System.Net.HttpWebResponse rep = (System.Net.HttpWebResponse) base.GetWebResponse(req);

                               if (rep.Headers["Set-Cookie"] != null)

                                {

                                       cookie = rep.Headers["Set-Cookie"];

                                }

                                return rep;

			}

                        public ClientService()

                        {

                                // Change the IP address pointing to netscaler.

                                this.Url = "http://10.102.4.111:18000";

                        }

                        public ClientService(string servername)

                        {

                                this.Url = "http://"+servername+"/soap";;

                        }

                }

	

	class rewrite 

	{

 		static ClientService client=null;

 

		[STAThread]

		static void Main(string[] args)

		{

			if ( args.Length < 3)

			{

				Console.WriteLine("Usage: getConfig <NS IP> username password");

				return;

			}

			string serverip = args[0];

			string username = args[1];

			string password = args[2];

			try {

          			Console.WriteLine("\nConnecting to server "+serverip+" ............\n");

				client = new ClientService(serverip);



				client.CookieContainer = new System.Net.CookieContainer();

				simpleResult result = client.login(username,password) ;

				Console.WriteLine("login : "+result.message);

       

				result = client.enablensfeature(featureEnum.REWRITE);

				Console.WriteLine("enablensfeature	REWRITE : " + result.message);



				result = client.addrewriteaction("rewrite_act1",actionrewritetypeEnum.DELETE,"HTTP.RES.HEADER(\"Server\").VALUE(0)",null,yesnoEnum.VALNOTSET);

				Console.WriteLine("addrewriteaction	rewrite_act1 : " + result.message);



				result = client.addrewritepolicy("rewrite_pol1","HTTP.RES.HEADER(\"Server\").CONTAINS(\"IIS\")","rewrite_act1",null);

				Console.WriteLine("addrewritepolicy	rewrite_pol1 : " + result.message);



				result = client.bindrewriteglobal_policy("rewrite_pol1",1,"END",rwglobalbindpointEnum.VALNOTSET,false,policylabelinvoketypeEnum.VALNOTSET,null);

				Console.WriteLine("bindrewriteglobal_policy	rewrite_pol1 : " + result.message);



				result = client.logout();

				Console.WriteLine("logout : " + result.message); 

			}

			catch (Exception ex) {

			      Console.WriteLine( "Exception: "+ex.ToString());

			}

		}



	}

	}

